### DISCUSSION TITLE
Future of Rocky Linux

### DISCUSSION THREAD

**Discussion by petiepooo** [Security-Onion-Solutions/securityonion#10650] [2.4] [answered]
RedHat has recently made a decision regarding the availability of their public source code releases.  They've essentially killed CentOS by acquisition, and are now trying to kill other no-cost RHEL clones that rose in response, such as Rocky and Alma Linux.
https://www.redhat.com/en/blog/furthering-evolution-centos-stream
https://news.itsfoss.com/red-hat-restricts-source-code/
https://www.phoronix.com/news/Red-Hat-CentOS-Stream-Sources

While Rocky Linux (RESF) has expressed their optimism, they have yet to announce a solution for moving forward.  There is no guarantee that there will be a legal solution, or that RedHat will stop in their attempts to eliminate no-cost RHEL clones.
https://rockylinux.org/news/2023-06-22-press-release/

I understand Security Onion Solutions has a goal of making this platform an easily supported homogenous appliance with a financially viable business around it.  Constant churn on the underlying operating system could hamper that.

The biggest advantage Docker CE gives a Linux administrator is its agnosticism concerning the underlying user space.  It is only your setup scripts and support policies that tie Security Onion 2.4 to a specific operating system, at least to my knowledge.  If there are technical limitations, I would surely be interested to hear about them.

I urge you to reconsider support of the Ubuntu LTS series.  While Canonical is also doing things I personally don't agree with (*cough* Snap!), I cannot argue with the success I've experienced running Security Onion on Ubuntu Server LTS platforms over the past 8+ years.  There are multiple systems that I have updated from Security Onion 14.04 to 16.04 or from 16.04 to Ubuntu 18.04 and v2.3 and now on to 20.04, all with just SSH access over a VPN; no console access via iLO/iDRAC/IPMI needed.  Upgrades to the next LTS version are a simple matter in Ubuntu, and your use of SaltStack and Docker makes it easy to restore Security Onion's services after a release upgrade.
https://ubuntu.com/server/docs/upgrade-introduction
https://docs.securityonion.net/en/2.3/appendix-b.html

It's been a few days of radio silence from Security Onion Solutions, so I thought I'd bring up the elephant in the room.  Given the recent announcement from RedHat, what are your intentions regarding its impact on Security Onion 2.4?

**Comment by dougburks** [comment, marked as answer]
https://blog.securityonion.net/2023/06/red-hat-rocky-linux-and-security-onion.html

**Reply by petiepooo** [reply]
Thank you.  Any details you can share about those contingency plans?  Us users may need to plan for them too.

I'm hoping you at least delay general availability of v2.4 and EOL of v2.3 until RESF has a concrete plan for moving forward and maybe even demonstrates it through a minor release or patch release from RHEL.

**Reply by dougburks** [reply]
The Rocky Linux team published another announcement over the weekend (I've added this to our blog post above):
https://rockylinux.org/news/brave-new-world-path-forward/

This gives us even more optimism that we'll be able to continue with our plans to use Rocky Linux.

**Reply by petiepooo** [reply]
A brief history of CentOS and RedHat for those new to the discussion.

- In 1994, Marc Ewing creates his own distribution of Linux which he names RedHat.  Bob Young buys Marc's company the following year and forms RedHat Software.

- In 1999, RedHat goes public, making enough to acquire Cygnus Solutions, a company providing commercial support for free software.

- In 2004, CentOS forks version RHL2.1AS (later renamed to RHEL) and creates patches stripping RedHat-specific trademarks and branding, offering a no-cost bug-for-bug compatible distro with no support.

- In 2006, Oracle announces Oracle Enterprise Linux, advertising a 100% RedHat-compatible kernel (RHCK) using RedHat's sources and undercutting RedHat's commercial support prices.

- In 2011, RedHat changes how they distribute their kernel source, moving to a single tarball.  A RedHet developer confirms in an interview that it was absolutely intended to make it more difficult for non-RedHat entities to build bug-for-bug compatible versions of their kernel.  https://lwn.net/Articles/430098/

- In 2014, RedHat captures the CentOS project.  The effect (and assumed intent) is to discourage use of other compatible distros (such as Oracle Linux and Scientific Linux), leaving CentOS as the gold-standard, no-cost, RHEL-compatible distro.  https://www.redhat.com/en/about/press-releases/red-hat-and-centos-join-forces

- In 2018, IBM announces its intent to buy RedHat.  The deal closes in 2019.  The community frets about IBM's track record as a steward of FOSS technology.  https://www.redhat.com/en/about/press-releases/ibm-closes-landmark-acquisition-red-hat-34-billion-defines-open-hybrid-cloud-future

- In 2020, IBM/RedHat announces an early end to CentOS 8, touting CentOS Stream as its replacement.  The community's fears confirmed, a few alternatives appear, such as Rocky Linux and Alma Linux.  https://blog.centos.org/2020/12/future-is-centos-stream/  https://www.tecmint.com/redhat-based-linux-distributions/

- In 2023, IBM/RedHat announces that RHEL source will no longer be posted publicly, available only under a support contract that prohibits further sharing.  This complicates other distro's ability to provide bug-for-bug compatible kernel builds.  https://www.redhat.com/en/blog/furthering-evolution-centos-stream

- In 2024, what do we think will happen?  Or will they wait for 2025 or 2026 before springing their next move on us?

IBM/RedHat will not stop trying to kill bug-for-bug compatible builds of their RHEL kernel.  In 2014, they embraced and extended their main competition and are now working on extinguishing them completely.  Rocky Linux and others may have found a way around it -- for now, but IBM wants your support contracts, and they are calculating, persistent, and litigious.  This will not be their last attempt to increase their RHEL revenues by any means at their disposal.

If you have, as you say, a fallback option, I take that to mean that there is nothing inherently tying you to a RHEL9-compatible distro.  That is, after all, one of the strengths of containerizing applications.

I don't share your optimism.  Rocky Linux's ability to continue is at odds with IBM's corporate goal, and Big Blue has better cards and more chips.  I believe now is the time for you to execute that fallback.

I'm sorry this has happened.  I don't like the position this has put you (and us) in.  Good luck.

**Reply by petiepooo** [reply]
Addendum:  Mike McGrath, VP of Core Platforms Engineering at RedHat, responds to the hate: https://www.redhat.com/en/blog/red-hats-commitment-open-source-response-gitcentosorg-changes

He's explicit in what he feels about Rocky Linux and the like:
> Ultimately, we do not find value in a RHEL rebuild and we are not under any obligation to make things easier for rebuilders; this is our call to make.

But my favorite part is this tidbit (second to last paragraph):
> Simply rebuilding code, without adding value or changing it in any way, represents a real threat to open source companies everywhere.

In my eyes, that viewpoint is the real threat -- not only to Rocky Linux, but to FOSS in general.

**Comment by Acewiza** [comment]
Rockies found out about it coming a few months ago, and nobody's happy.  It won't be long before there's no downstream distros anybody can trust, Ubuntu included.

**Comment by Acewiza** [comment]
[Little back story.](https://www.jeffgeerling.com/blog/2023/im-done-red-hat-enterprise-linux)
